tools/serverpod_cli/lib/src/analyzer/models/entity_dependency_resolver.dart

```diff
@@ -30,6 +33,46 @@ class ModelDependencyResolver {
     });
   }
 
+  static void _resolveInheritance(
+    ClassDefinition classDefinition,
+    List<SerializableModelDefinition> modelDefinitions,
+  ) {
+    if (classDefinition.extendsClass != null) {
+      try {
+        var baseClass = modelDefinitions
+            .whereType<ClassDefinition>()
+            .firstWhere((element) =>
+                element.className == classDefinition.extendsClass?.className);
+
+        classDefinition.extendsClass = ExtendsClassInheritanceDefinition(
+          baseClass.className,
+          baseClass.fields,
+          p.joinAll([
+            ...baseClass.subDirParts,
+            '${baseClass.fileName}.dart',
+          ]),
+        );
+      } catch (e) {
+        classDefinition.extendsClass = null;
+      }
+
+      try {
+        List<ClassDefinition?> subClasses = modelDefinitions
+            .whereType<ClassDefinition>()
+            .where((element) =>
+                element.extendsClass?.className == classDefinition.className)
+            .toList();
+
+        classDefinition.subClasses = BaseClassInheritanceDefinition(
+          classDefinition.className,
+          subClasses,
+        );
+      } catch (e) {
+        classDefinition.subClasses = null;
```

Comment:
Suggestion: Same as above, preserve the Unresolved if we are not able to resolve the reference.